backend/src/server.ts

```diff
@@ -0,0 +1,26 @@
+import express, { Express, Request, Response } from 'express';
+import bodyParser from 'body-parser';
+import 'dotenv/config';
+import Logger from '@utils/logger';
+import authRouter from '@routes/auth.route';
+import mongoose from 'mongoose';
+
+const app: Express = express();
+app.use(bodyParser.json());
+app.use('/auth', authRouter);
```

Comment:
Create an applicationRouter.ts which contains all of these routes. SOLID principles, each class has a single responsibility. 

```suggestion
app.use(applicationRouter);
```

Also create a baseRouter middleware to parse JSON data instead of app.use(bodyParser.json())

```

// base.router.ts
import express from 'express';
import { Router } from 'express';
import bodyParser from 'body-parser';

const baseRouter: Router = express.Router();

// Define common middleware
baseRouter.use(bodyParser.json());

export default baseRouter;

```